src/measure_theory/measure/finite_measure_weak_convergence.lean

```diff
@@ -0,0 +1,246 @@
+  exact ennreal.lt_top_iff_ne_top.mp (μ.prop).measure_univ_lt_top,
+end
+
+-- TODO: Another alternative would use a bijection with `vector_measure α ℝ≥0`.
+instance : add_comm_monoid (finite_measure α) :=
+(finite_measure.coe_injective).add_comm_monoid
+  (coe : finite_measure α → measure α) finite_measure.coe_zero finite_measure.coe_add
+
+/-- Coercion is an `add_monoid_hom`. -/
+@[simps]
+def coe_add_monoid_hom : finite_measure α →+ measure α :=
+{ to_fun := coe, map_zero' := coe_zero, map_add' := coe_add }
+
+-- TODO: Another alternative would use a bijection with `vector_measure α ℝ≥0`.
+instance {α : Type*} [measurable_space α] : module ℝ≥0 (finite_measure α) :=
+function.injective.module _ coe_add_monoid_hom finite_measure.coe_injective coe_smul
+
+end finite_measure -- end namespace
+
+end finite_measure -- end section
+
+section probability_measure
+/-!
+### Probability measures
+In this section, we define the `Type` of probability measures on a given measurable space. This
+type will be equipped with the topology of weak convergence of measures when the measurable space
+is a topological space equipped with its Borel sigma-algebra.
+
+There is a coercion (embedding) to finite measures on the same space.
+-/
+
+variables {α : Type} [measurable_space α]
+
+/-- Probability measures are defined as the subtype of measures that have the property of being
+probability measures (i.e., their total mass is one). -/
+def probability_measure (α : Type) [measurable_space α] : Type :=
+{μ : measure α // is_probability_measure μ}
+
+namespace probability_measure
+
+instance [inhabited α] : inhabited (probability_measure α) :=
+{ default := ⟨measure_theory.measure.dirac (default α),
+              measure_theory.measure.dirac.is_probability_measure⟩ }
+
+/-- A probability measure can be interpreted as a measure. -/
+instance : has_coe (probability_measure α) (measure_theory.measure α) := coe_subtype
+
+instance : has_coe_to_fun (probability_measure α) :=
+⟨λ _, set α → ℝ≥0, λ μ s, (μ s).to_nnreal⟩
+
+instance to_measure.is_probability_measure (μ : probability_measure α) :
+  is_probability_measure (μ : measure α) := μ.prop
+
+lemma to_fun_eq_to_measure_to_nnreal (ν : probability_measure α) :
+  (ν : set α → ℝ≥0) = λ s, ((ν : measure α) s).to_nnreal := rfl
+
+@[simp] lemma val_eq_to_measure (ν : probability_measure α) : ν.val = (ν : measure α) := rfl
+
+lemma coe_injective : function.injective (coe : probability_measure α → measure α) :=
+subtype.coe_injective
+
+@[simp]
+lemma to_fun_univ (ν : probability_measure α) : ν univ = 1 :=
+begin
+  rw [to_fun_eq_to_measure_to_nnreal, ←ennreal.one_to_nnreal],
+  exact congr_arg ennreal.to_nnreal ν.prop.measure_univ,
+end
+
+instance has_coe_to_finite_measure : has_coe (probability_measure α) (finite_measure α) :=
+{ coe := λ μ , ⟨μ, infer_instance⟩ }
+
+/-- A probability measure can be interpreted as a finite measure. -/
+def to_finite_measure (μ : probability_measure α) : (finite_measure α) := μ
```

Comment:
I was playing with this a bit, and had the impression that, with the coercion
```
has_coe (probability_measure α) (finite_measure α)
```
in place, tricky things sometimes happen (for example, coercion from `probability_measure α` to `measure α` seems sometimes to follow the path via `finite_measure α` rather than the direct path.

I didn't write down a specific example of this happening, though I can try to reconstruct it.  Did you ever notice anything like this?

I'd incline to avoid it by _not_ making this particular route a coercion -- i.e., provide
```suggestion
/-- A probability measure can be interpreted as a finite measure. -/
def to_finite_measure (μ : probability_measure α) : (finite_measure α) := ⟨μ, infer_instance⟩
```
but no associated coercion.

Reply:
This makes a lot of sense now that you say it. I think I will actually want to be explicit about whenever this mapping is used.

Thank you!